test: remove duplicated ban test

brunoerg opened the pull request:
Test the ban list is preserved through restart has been done by both `rpc_setban` and `p2p_disconnect_ban`. Since `p2p_disconnect_ban` does it in a more elegant way, we can keep only it and remove the other one.

https://github.com/bitcoin/bitcoin/blob/bf1b6383dbbfdd0c96a161d4693a48bf3a6b6150/test/functional/p2p_disconnect_ban.py#L74-L110

brunoerg committed:
test: remove duplicated ban test

Test the ban list is preserved through restart has been
done by both `rpc_setban` and `p2p_disconnect_ban`.
Since `p2p_disconnect_ban` does it in a more elegant
way, we can keep only it and remove the duplicated one.

tdb3 commented on test/functional/rpc_setban.py:
```diff
@@ -60,24 +60,6 @@ def run_test(self):
         assert not self.is_banned(node, tor_addr)
         assert not self.is_banned(node, ip_addr)
 
-        node.setban(tor_addr, "add")
-        assert self.is_banned(node, tor_addr)
-        assert not self.is_banned(node, ip_addr)
```
Seems like this would also remove testing of banning an onion address, which doesn't seem ideal. Would recommend keeping this.

brunoerg replied:
Done.

tdb3 commented on test/functional/rpc_setban.py:
```diff
@@ -60,24 +60,6 @@ def run_test(self):
-        assert self.is_banned(node, tor_addr)
-        assert not self.is_banned(node, ip_addr)
-
-        node.setban(tor_addr, "remove")
-        assert not self.is_banned(self.nodes[1], tor_addr)
-        assert not self.is_banned(node, ip_addr)
```
Same for here.  If this were kept, it would continue to test ban removal for onion addresses.

brunoerg replied:
Done.

tdb3 reviewed (commented):
Nice work taking a more holistic view of test coverage (avoiding duplicates).

Inline comments provided.  In a nutshell, unless it's covered elsewhere (and I'm missing something), I would recommend keeping lines focused on testing banning/unbanning onion addresses.  Looks like the removal of restarting/checking lines can be done while keeping onion address banning/unbanning/checking.

This would prevent the preceding lines from false advertising:
https://github.com/bitcoin/bitcoin/blob/9cbfb82a610c3a40777ba372588564810e736d7c/test/functional/rpc_setban.py#L56-L61

brunoerg commented:
Thanks, @tdb3 for reviewing it. Just changed it to keep the ban/unban of an onion addresses and moved the test persistence across node restart to `p2p_disconnect_ban`.

tdb3 commented on test/functional/p2p_disconnect_ban.py:
```diff
@@ -87,7 +88,7 @@ def run_test(self):
 
         # Move time forward by 3 seconds so the third ban has expired
```
nit:
```diff
-        # Move time forward by 3 seconds so the third ban has expired
+        # Move time forward by 3 seconds so the forth ban has expired
```

brunoerg replied:
Done

tdb3 commented on test/functional/p2p_disconnect_ban.py:
```diff
@@ -77,6 +77,7 @@ def run_test(self):
         self.nodes[1].setmocktime(old_time)
         self.nodes[1].setban("127.0.0.0/32", "add")
         self.nodes[1].setban("127.0.0.0/24", "add")
+        self.nodes[1].setban("pg6mmjiyjmcrsslvykfwnntlaru7p5svn6y2ymmju6nubxndf4pscryd.onion", "add")
```
nit:  The line just after this has an index that isn't adjusted, but still passes.
```python
assert_equal("192.168.0.1/32", listBeforeShutdown[2]['address'])
```
At first glance, this might be due to how the address types are organized in the list returned by `listbanned()` but if this organization changes then the test could fail in the future.  Recommend this be updated to increase robustness.  For example:
```python
addrs_before_shutdown = [a['address'] for a in self.nodes[1].listbanned()]
assert "192.168.0.1/32" in addrs_before_shutdown
```
Author's call as to whether this is part of this PR or a new/future one.  There are other parts of test that depend on specific ordering in the list (this isn't the only spot).

brunoerg replied:
I'll leave it as it for now. Perhaps we could change it if/when when the behaviour of `listbanned` changes.

tdb3 commented on test/functional/p2p_disconnect_ban.py:
```diff
@@ -87,7 +88,7 @@ def run_test(self):
+        assert_equal(len(self.nodes[1].listbanned()), 5)
 
         self.log.info("Test ban_duration and time_remaining")
         for ban in self.nodes[1].listbanned():
```
Recommend adding checks for the onion address in this loop.

Also recommend adding check for the onion address the after shutdown (technically after restart) here:
https://github.com/bitcoin/bitcoin/blob/605b1a84093f38a328940f96fcdc35866f5a7422/test/functional/p2p_disconnect_ban.py#L107-L111

brunoerg replied:
Nice idea, done!

tdb3 reviewed (commented):
Pulled, built, and ran all unit and functional tests (all passed).
Updated comments for new commit 605b1a84093f38a328940f96fcdc35866f5a7422.

brunoerg commented:
Force-pushed addressing https://github.com/bitcoin/bitcoin/pull/29688#discussion_r1536966972

tdb3 commented on test/functional/p2p_disconnect_ban.py:
```diff
@@ -85,9 +86,9 @@ def run_test(self):
+        assert_equal(len(self.nodes[1].listbanned()), 5)
 
         self.log.info("Test ban_duration and time_remaining")
         for ban in self.nodes[1].listbanned():
```
Looking good!  What if we button things up by adding checks for the onion address inside the for loop `for ban in self.nodes[1].listbanned():`?

https://github.com/bitcoin/bitcoin/blob/7cdef4bc1148a6641ad6cfb2ade90cbe8fba82cb/test/functional/p2p_disconnect_ban.py#L93-L103

brunoerg replied:
Nice idea, done!

tdb3 reviewed (commented):
Inline comment left for 7cdef4bc1148a6641ad6cfb2ade90cbe8fba82cb.
Pulled and re-ran `p2p_disconnect_ban` (passed).

brunoerg commented:
Force-pushed addressing https://github.com/bitcoin/bitcoin/pull/29688#discussion_r1538456073

tdb3 reviewed (commented):
ACK for e30e8625bbc42045b8b757a8d7e80c20cc61cebf.
Comments addressed.  Pulled latest commit and ran `p2p_disconnect_ban` (v1 and v2 variants).  Passed.

hernanmarino reviewed (approved):
tested ACK e30e8625bbc42045b8b757a8d7e80c20cc61cebf

BrandonOdiwuor reviewed (approved):
ACK e30e8625bbc42045b8b757a8d7e80c20cc61cebf

Looks good to me

alfonsoromanz reviewed (approved):
ACK e30e8625bbc42045b8b757a8d7e80c20cc61cebf

achow101 commented:
It seems like these two tests primarily revolve around the `setban` RPC and do a couple of the same things, so perhaps they can be consolidated into one test?

brunoerg commented:
> It seems like these two tests primarily revolve around the `setban` RPC and do a couple of the same things, so perhaps they can be consolidated into one test?

See #26863. Closed due to lack of interest.

achow101 commented:
ACK e30e8625bbc42045b8b757a8d7e80c20cc61cebf
